File: python/collatz-conjecture/collatz_conjecture.py
def steps(number):
    if number <= 0:
        raise ValueError("invalid input, must be greater than 0")

    count = 0
    while number != 1:
        number = number // 2 if number % 2 == 0 else 3 * number + 1
        count += 1

    return count

File: cpp/collatz-conjecture/collatz_conjecture.cpp
#include "collatz_conjecture.h"
#include <stdexcept>

namespace collatz_conjecture {
int steps(int num) {
    if (num <= 0) {
        throw std::domain_error("invalid input");
    }

    int count = 0;
    while (num != 1) {
        if (num % 2 == 0) {
            num /= 2;
        } else {
            num = num * 3 + 1;
        }
        count++;
    }
    return count;
}
} // namespace collatz_conjecture
